File: Problem_set_1/Introductory_problems/p04_Increasing_Array/IncreasingArray.ts
const input = require("prompt-sync")();

const size_array: number = Number(input())

const num_arr: number[] = input().split(" ").map(Number);

let totalMoves: number = 0;

for (let i: number = 1; i < size_array; i++) {
  const previous = num_arr[i - 1]!;
  const current = num_arr[i]!;

  if (current < previous) {
    const gap = previous - current;
    totalMoves += gap;
    num_arr[i] = previous;
  }
}

console.log(totalMoves);

export {};

File: Problem_set_1/Introductory_problems/p04_Increasing_Array/increasingArray.kt
fun main(){
    val sizeArray: Int = readln().toInt()
    var numArray: IntArray = IntArray(sizeArray)

    for(i in 0 until numArray.size){
        numArray[i] = readln().toInt()
    }

    var totalMoves: Long = 0;
    for(i in 1 until numArray.size){
        val current = numArray[i]
        val previous = numArray[i - 1]

        if(current < previous){
            val gap = previous - current
            totalMoves += gap
            numArray[i] = previous
        }
    }

    println(totalMoves)
}